### PR TITLE
Adkernel review

### PR DESCRIPTION
**Pull request by snahornyi**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java [hunk 1]
```diff
@@ -104,25 +108,22 @@ private ExtImpAdkernel parseAndValidateImpExt(Imp imp) {
...
                 .audio(null)
                 .xNative(null);
         return imp.getBanner() != null ? impBuilder.video(null).build() : impBuilder.build();
     }
 
     private static boolean hasNoImpressions(Map<ExtImpAdkernel, List<Imp>> pubToImps) {
         return pubToImps.values().stream()
-                .mapToLong(Collection::size)
-                .sum() == 0;
+                .noneMatch(imp -> imp.size() > 0);
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:126]: return pubToImps.values().stream().allMatch(CollectionUtils::isEmpty);

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java [hunk 2]
```diff
@@ -145,8 +146,8 @@ private static boolean hasNoImpressions(Map<ExtImpAdkernel, List<Imp>> pubToImps
                 .build();
     }
 
-    private static BidRequest createBidRequest(
-            List<Imp> imps, BidRequest.BidRequestBuilder requestBuilder, Site site, App app) {
+    private static BidRequest createBidRequest(List<Imp> imps, BidRequest.BidRequestBuilder requestBuilder,
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:149]:
```
    private static BidRequest createBidRequest(List<Imp> imps,
                                               BidRequest.BidRequestBuilder requestBuilder,
                                               Site site,
                                               App app)
```

File: src/test/java/org/prebid/server/bidder/adkernel/AdkernelBidderTest.java [hunk 3]
```diff
@@ -70,8 +70,9 @@ public void makeHttpRequestsShouldReturnErrorWhenImpHasNoBannerOrVideo() {
 
         // then
         assertThat(result.getValue()).isNull();
-        assertThat(result.getErrors()).hasSize(1)
-                .containsOnly(BidderError.badInput("Invalid imp id=123. Expected imp.banner or imp.video"));
+        assertThat(result.getErrors())
+                .allMatch(error -> error.getType() == BidderError.Type.bad_input
+                        && error.getMessage().startsWith("Invalid imp id=123. Expected imp.banner or imp.vide"));
```
Comment (DGarbar) [inline src/test/java/org/prebid/server/bidder/adkernel/AdkernelBidderTest.java:75]: I don't see why it is better (in all cases)

**Review by DGarbar — CHANGES_REQUESTED** [review]

**Review by DGarbar — APPROVED** [review]

**Review by rpanchyk — COMMENTED** [review]
LGTM, just merge master.

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**